Frontend new UI features and appearance [DO NOT MERGE]

nikhilxb opened the pull request:
New features introduced include:

- Inline bundle details expansion panel
- Inline New Upload / New Run expansion panel
- Simplified Worksheet main display view
- Inserting bundles at a particular location within worksheet

percyliang reviewed (changes requested):
Great!  General comments:
- There are some big functions containing a deep hierarchy of components - can we break these up so that it's easier to see the big picture?
- We should add high-level documentation outlining what the constituent components (what semantically the main content represents versus the side bar).
- There's a bunch of low-level indentation / formatting problems that we should try to tidy up.

percyliang commented:
Don't merge this until we've user tested it.

epicfaace committed:
fix: make sure upload_to_bundle_store only writes to binary files

epicfaace committed:
fix: properly print binary contents to stdout.buffer

epicfaace committed:
Merge master

epicfaace committed:
fix: black format

epicfaace committed:
use xenial

epicfaace committed:
Revert "use xenial"

This reverts commit 7842bde12a54c84a9621a4cffccc61445cccdcbf.

epicfaace committed:
try flushing stdout

epicfaace committed:
fix: fix binary in worker_model

epicfaace committed:
fix: remove unnecessary decode

epicfaace committed:
feat: set max request params to 0 by default (python 2 allows comparing numbers with None but not python 3)

epicfaace committed:
fix: load json string

epicfaace committed:
ci: try changing travis script

epicfaace committed:
fix: show errors on test

epicfaace committed:
fix: fix typo

epicfaace committed:
Revert "ci: try changing travis script"

This reverts commit 2394d1d3ae27446437a8f753a6f0ad64014ed992.

epicfaace committed:
fix: add try-catch to checkin

epicfaace committed:
fix: json.dump to string-writable file

epicfaace committed:
fix: open files in rb mode for path_contents

epicfaace committed:
fix (tests): create binary versions of path_contents and run_command

epicfaace committed:
fix(test): more binary output fixes

epicfaace committed:
fix: properly encode POST data

epicfaace committed:
fix: add charset=utf8 to all db urls

epicfaace committed:
fix: use integer division

epicfaace committed:
fix: don't encode strings when returning uid

epicfaace committed:
fix: fix invalid sqlite url

epicfaace committed:
fix: fix binary/unicode issues for netcat tests

epicfaace committed:
chore: black format

epicfaace committed:
fix: re-add truncating of console output

epicfaace committed:
chore: remove double parenthesis around print statements

epicfaace committed:
fix: properly truncate output

epicfaace committed:
chore: remove duplicate import

epicfaace committed:
chore: keep iterators through .items()

epicfaace committed:
chore: remove duplicate imports

epicfaace committed:
chore: change list(range to range

epicfaace committed:
chore: convert most list()'s to iterators

epicfaace committed:
fix: syntax error

epicfaace committed:
chore: remove more iter's

epicfaace committed:
chore: remove extra print statement

epicfaace committed:
chore: remove flush

epicfaace committed:
fix: replace eval(input()) with input()

epicfaace committed:
Use binary string

epicfaace committed:
Update tests/worker/file_util_test.py

epicfaace committed:
Update worker/codalabworker/worker.py

epicfaace committed:
Update worker/codalabworker/local_run/local_run_manager.py

epicfaace committed:
Update worker/codalabworker/local_run/local_dependency_manager.py

epicfaace committed:
Update test_cli.py

epicfaace committed:
Update worker/codalabworker/file_util.py

epicfaace committed:
Update setup.py

epicfaace committed:
Update worker/setup.py

epicfaace committed:
chore: replace list() with iterators

epicfaace committed:
fix: force python 3.6

epicfaace committed:
fix: store login cookie as json, not a pickled object

This is mainly due to problems with Python 3 reading existing Python
2-pickled cookies. Additionally, there are security issues with storing
pickles as cookies and bottle recommends us to store JSON objects anyway
(https://github.com/bottlepy/bottle/issues/900).

percyliang committed:
Merge branch 'frontend' of github.com:/codalab/codalab-worksheets into frontend

percyliang committed:
Merge branch 'master' into frontend

percyliang committed:
Merge branch 'master' into frontend

epicfaace committed:
Merge

epicfaace committed:
fix: run tests with right version of python

epicfaace committed:
fix: print test output to stdout in real time

epicfaace committed:
chore: reformat black

epicfaace committed:
ci: set travis python encoding to utf8

epicfaace committed:
fix:  run competitiond with python3.6

epicfaace committed:
fix: print ascii for travis

epicfaace committed:
Merge remote-tracking branch 'origin/master' into python3

epicfaace committed:
fix (unicode): properly encode subprocess args before sending to command line

epicfaace committed:
fix: make sure fileobj passed to _upload_with_chunked_encoding is bytes

epicfaace committed:
fix: remove unicode restrictions

epicfaace committed:
fix: remove unicode_constructor

epicfaace committed:
Revert "fix: remove unicode_constructor"

This reverts commit ced5644aed5edc8c08e7bf6c5d1b7a86ab836921.

epicfaace committed:
update description for unicode_constructor

epicfaace committed:
fix: print only ascii in tests

epicfaace committed:
fix: properly accept unicode arguments on the command line

epicfaace committed:
chore: doc reformat

epicfaace committed:
fix: allow unicode in metadata

epicfaace committed:
Merge

epicfaace committed:
fix: add more unicode tests

epicfaace committed:
fix: don't require ascii on frontend

epicfaace committed:
fix: save test file in utf-8 encoding

epicfaace committed:
fix: make sure python encoding for server file is utf-8

epicfaace committed:
fix: open file correctly with with statement

epicfaace committed:
fix: allow unicode bodies

epicfaace committed:
test: add wedit tests for unicode

epicfaace committed:
remove non-ascii error handling

epicfaace committed:
reformat with black

epicfaace committed:
Merge remote-tracking branch 'origin/master' into python3

epicfaace committed:
fix: use python3 pip

epicfaace committed:
Revert "fix: use python3 pip"

This reverts commit 6ce7b316ce437a465e82dd7b4fcb85a41f3d3776.

epicfaace committed:
fix: change docs to be python 3 again

epicfaace committed:
use python3 pip

epicfaace committed:
try ENV LANG for python encoding

epicfaace committed:
remove PYTHONIOENCODING from travis env

epicfaace committed:
3.5 to 3.6

epicfaace committed:
codalabworker requires 3.6

epicfaace committed:
Merge

epicfaace committed:
fix: fix print

epicfaace committed:
black format

epicfaace committed:
fix typo

epicfaace committed:
fix: upgrade sqlalchemy

epicfaace committed:
fix: package_data values should be array

epicfaace committed:
downgrade sqlalchemy

epicfaace committed:
add test for unicode body

epicfaace committed:
Merge remote-tracking branch 'origin/master' into python3

epicfaace committed:
use 2to3 on gen-cli-docs

epicfaace committed:
fix gen-rest-docs and gen-cli-docs

epicfaace committed:
remove collation from mysql url

epicfaace committed:
add bundle model test for saving metadata in unicode

epicfaace committed:
black format

epicfaace committed:
test: test actual mysql and sqlite bundle models

epicfaace committed:
fix bundlemodel test

epicfaace committed:
fix: actually fix bundle_model_test

epicfaace committed:
Merge

epicfaace committed:
fix: use utf8mb4_unicode_ci collation

epicfaace committed:
specify collation properly

epicfaace committed:
fix: fix collation for worksheet item value and tag names

epicfaace committed:
black format

epicfaace committed:
Update worker/codalabworker/rest_client.py

epicfaace committed:
Merge

epicfaace committed:
use unicode and unicodetext

epicfaace committed:
black format

epicfaace committed:
fix: decode strings in hacky way

epicfaace committed:
fix: import array

percyliang committed:
merge

percyliang committed:
Merge branch 'master' into frontend

epicfaace committed:
fix: fix worksheet_tags test

epicfaace committed:
Merge

epicfaace committed:
remove sqlite test for now

epicfaace committed:
remove encode and decode

epicfaace committed:
fix: go back to python-2 way

epicfaace committed:
fix: get a string branch

epicfaace committed:
replace python with python3.6

epicfaace committed:
remove comments

epicfaace committed:
fix: revert bundle_model tests, we  don't actually need them

epicfaace committed:
show help with "cl" argument, fixes #1332

epicfaace committed:
fix: allow unicode in worksheet tags

epicfaace committed:
test: add test for unicode in worksheet title

epicfaace committed:
remove bundle_model_test

epicfaace committed:
fix: unicode in metadata

epicfaace committed:
black format

epicfaace committed:
Revert tables.py

epicfaace committed:
attempt to add unicodecoerce custom types

epicfaace committed:
remove metadata_row check

epicfaace committed:
remove encode_str and decode_str

epicfaace committed:
re-add bundle_model_test

epicfaace committed:
add bundlemodeltest fix indent

epicfaace committed:
remove unicode support for title, description, tags

epicfaace committed:
fix --title wedit test

epicfaace committed:
add decode_str

epicfaace committed:
black format

epicfaace committed:
Remove if bytes statement

epicfaace committed:
Open json files using r mode instead of rb mode

epicfaace committed:
fix: remove unicode argument hack, because default encoding has been properly set in Dockerfile

epicfaace committed:
fix: remove unneeded try-except

epicfaace committed:
support binary args in tests

epicfaace committed:
remove unneeded try-catch

epicfaace committed:
black format

epicfaace committed:
Delete types.py

epicfaace committed:
Merge branch 'python3' of github.com:codalab/codalab-worksheets into python3

epicfaace committed:
feat: sticky header

epicfaace committed:
fix: load css through react-scripts, not through static html

epicfaace committed:
improve styling of worksheet header

epicfaace committed:
fix delete button

epicfaace committed:
Merge

epicfaace committed:
black format

epicfaace committed:
fix python -> python3.6

epicfaace committed:
fix gen-rest-docs

epicfaace committed:
Merge

epicfaace committed:
Merge

epicfaace committed:
Merge

epicfaace committed:
Merge

epicfaace committed:
Frontend - fix crashing when selecting rows (#1421)

* remove numTableRows check in scrollToItem

* re-add condition, but make it better

epicfaace committed:
Add upload buttons to navbar (#1424)

* remove numTableRows check in scrollToItem

* re-add condition, but make it better

* feat: move upload buttons to navbar

* fix: hide run/text dialog when focus is changed

epicfaace committed:
Add web-cli back to frontend (#1434)

* re-add web cli

* fix: show all lines of terminal initially

* Remove instances of #worksheet_panel

* keep search-hidden class to remove big empty space at top of worksheet

* fix: make web cli sticky

* fix: allow for showing / hiding of action bar

* Revert "Remove web terminal shortcuts / verbiage on help screen (#1245)"

This reverts commit 6fb336cb4c1f6eef4b98f2da8a5b26f3a33008ab.

yipenghe committed:
When a item is selected, the highlight color now occupies whole row (#1466)

* When a item is selected, the highlight color now occupies whole row

* Update frontend/src/css/imports.css

yipenghe committed:
Add expand button to control bundle row detail showing (#1474)

* adding expand button

* positon the expand button

* unindent

* remove extra space

* move button to left of row

* remove console.log

* use tenary operator

yipenghe committed:
Change display of run time for bundles (#1475)

* Change display of run time for bundles

* Format run time better

* use existing render duration function'

* remove incorrect comment

* Add expand button to control bundle row detail showing (#1474)

* adding expand button

* positon the expand button

* unindent

* remove extra space

* move button to left of row

* remove console.log

* use tenary operator

* Change display of run time for bundles

* Format run time better

* use existing render duration function'

* remove incorrect comment

* simplify code block

* simplify code block

epicfaace committed:
fix crashing when deleting last bundle (#1488)

fixes #1208

yipenghe committed:
add blurred shadow edges (#1473)

Levilian committed:
dialog box has higher z-index over sticky worksheet title and cli (#1482)

* dialog box has higher z-index over sticky worksheet title and cli

* update z-index of tooltip

Levilian committed:
changed z-index of keyboard shortcuts (#1491)

yipenghe committed:
Deselect when clicking outside worksheet row items (#1447)

* deselect (unfocus) when clicking outside any item

* changing function name for clarity

* use safer way to deselect

* remove dummy event parameters to be consistent

* add back new line

* remove tap

locally there is no taps...

epicfaace committed:
fix mathjax causing crash in race case (#1489)

epicfaace committed:
fix: make contiguous blocks of text appear on different lines (#1507)

epicfaace committed:
Upgrade react to 16.8 (#1508)

yipenghe committed:
Add ErrorMessage component [WIP] (#1481)

* add draft ErrorMessage component

* remove extra lines

* change style, display uuid

* add draft ErrorMessage component

* remove extra lines

* change style, display uuid

* Change message & return error message for worksheets/

* font size change

Levilian committed:
Major update for worksheet header's look (#1510)

* Major update for worksheet header's look

1. standardized all the buttons and used grid for ordering
2. used material ui tooltip for keyboard shortcuts
3. enabled the toggling of web terminal from the header
4. combined worksheet information into one div

epicfaace committed:
fix worksheet insert (#1503)

epicfaace committed:
Allow adding items to empty worksheets (#1506)

* fix: append to end of worksheet if nothing is focused

* fix: hide text editor upon pressing "escape"

* fix: allow adding items to empty worksheets

* Update frontend/src/components/worksheets/WorksheetItemList.js

* Update frontend/src/components/worksheets/WorksheetItemList.js

yipenghe committed:
cleaning up the hovering insert panels (#1538)

yipenghe committed:
Don't open uuid link in bundle row by pressing 'enter'  (#1541)

* prevent the default behavior of opening selected links

* comment

* comment space

yipenghe committed:
Make bundle height similar to current version (#1540)

* shrink bundle height

* Position delete buttom better (#1539)

* add margin bottom for centering vertically

* add margin bottom for centering vertically

* adjust according to bundle row height change

* shrink bundle height

* Position delete buttom better (#1539)

* add margin bottom for centering vertically

* add margin bottom for centering vertically

* adjust according to bundle row height change

* word-wrap

epicfaace committed:
re-add context menu (#1554)

yipenghe committed:
Allow unicode in title (#1511)

* unicode in title & name

* remove console.log

* rebase to frontend

* remove mysql conf

* remove console.log

* save

* fix frontend side unicode checking: 1. use encodeURIcomponent 2. restore name restriction

* restore name check in worksheet.py

* extra line removal

* empty line before return

* use encode/decode_str

* remove extra line

* remove import urllib

* add null check

* update encode/decode_str comment, nit change

* update encode/decode_str comment

* change unicode test case test_cli

* remove extra line

Levilian committed:
changed the default disk space to empty (#1545)

* changed the default disk space to empty

* remove unused const

Levilian committed:
fix the text when item gets selected; beautified the box shadow (#1546)

* fix the text when item gets selected; beautified the box shadow

* added back blurry box and border-left to item; fixed moving item when selected

Levilian committed:
changed how worksheet title and  web cli looks (#1549)

* changed how worksheet title and  web cli looks
1520: made cli more opaque
1532: made navbar aboe the web cli, which is hidden by default
1548: changed how title and name look like and their respective hover behavior

* changed title color to blue; increased opacity of cli

yipenghe committed:
check_contains unicode test (#1564)

Levilian committed:
add highlight to the right row when adding/deleting new rows (#1559)

* add highlight to the right row when adding/deleting new rows

epicfaace committed:
fix: ctrl+enter to save bundle (#1561)

epicfaace committed:
Install react-dnd dependencies (#1565)

epicfaace committed:
fix scrolling with keyboard shortcuts (#1553)

* fix scrolling with keyboard shortcuts

* fix: don't do default scroll behavior when u/d pressed

* fix ws scrolling, move scrollbar to only worksheet (not whole page)

* fix: if nothing is selected, don't scroll

* fix: don't allow scrolling up above top cell

yipenghe committed:
Change bundle detail layout (#1543)

* draft new bundle detail

* updates

* updates

* update order of info

* make name field optional

* moved run time to side bar; redesigned main content

* remove dummy file contents

* tenary operator format

* uncomment test comments

* use classNames to reuse styles

* add back type script checks

* format

* add comment diviers

* draft new bundle detail

* updates

* updates

* update order of info

* make name field optional

* moved run time to side bar; redesigned main content

* remove dummy file contents

* tenary operator format

* uncomment test comments

* use classNames to reuse styles

* add back type script checks

* format

* add comment diviers

* Update BundleRow.js

* merge confict

* fix typo, wording

* alignment, reuse style

epicfaace committed:
fix: can't deselect a cell (#1560)

epicfaace committed:
Merge remote-tracking branch 'origin/master' into frontend

epicfaace committed:
Merge branch 'frontend' of github.com:codalab/codalab-worksheets into frontend

Levilian committed:
Add keyboard shortcuts for major actions (#1578)

* Add keyboard shortcuts for major actions
- insert text after current cell
- upload after current cell
- run after current cell
- enter for markdown editing
- ctrl+enter for saving
- backspace to delete
- fixed moving to beginning shortcut
- fixed refreshing worksheet focus

* fixed focus and shortcuts issues
- added del key to delete
- added esc to go back
- fixed focus bug on pressing keys in terminal
- added keyboard shortcut to focus the terminal
- implemented re-focus on delete, run, cell, and upload

Levilian committed:
fix mode property doesn't exist for item bug (#1596)

Levilian committed:
fixed no focus adding a new run bug

Levilian committed:
Revert "fixed no focus adding a new run bug"

This reverts commit e58b1b48e72f42fe14c7364ece24a465a183c75c.

Levilian committed:
fixed a bug where adding a new run without focus would crash (#1597)

yipenghe committed:
Bundle bulk operation on web ui (#1579)

> Supports bulk bundle operation from the web UI
> Can use 'd' for detach, 'backspace/del' for delete, 'v' for kill on multiple selected bundles, they open up a dialog for confirmation, can 'f' to select force deletion on deletion dialog, can use 'enter' to confirm action. It's the same as running cl rm, cl detach & cl kill on selected bundles
> show/hides bulk bundle action bar when at least one/none is selected, hides the worksheet action bar when bundles are selected (can still use shortcuts to perform)
> Most shortcuts are disabled when a dialog is opened
> Currently the checkmark get refreshed when a new bundle row is added / a bundle operation is successfully performed, the selection persists if bundle operation failed
> some minor changes to bundle row style

* support bulk delete, detach & kill

* support force delete in bulk operation

* add error dialog when action fails

* show/hide bundle operations when at least one/no bundle is selected

* mimic gmail behavior

* bind enter to confirm actions

* dont trigger things behind dialog

* add dismiss button for error messages

* More information about force deletion

* remove floating deletion on bundle row

* layout & shortcuts, disable worksheet/bundlerow shortcuts when a dialog is open, use x to select a bundle row, use space to select all rows in a table

* update tableitem control bundlerow checkstatus logic, fix deletion bug, added comments to document the mechanism

yipenghe committed:
Fix unicode error (#1630)

fix unicode error

yipenghe committed:
Styles bundle rows (#1639)

* add left blur bar to bundle rows on focus

* style bundle row

* add transparent border to content row class/ add border to header row for alignment

yipenghe committed:
fix autocomplete command (#1645)

epicfaace committed:
Refactor CSS (#1603)

* install sass

* refactor imports and Bundle scss

* fix: don't do a production build in dev mode

* upgrade react-scripts to support scss

* refactor out worksheet scss

* remove select2

* refactor bundle-state css

* refactor filebrowser css

* import bootstrap

* remove chatbox css

* refactor extraworksheethtml css

* fix filebrowser import

* properly use bootstrap, refactor terminal

* fix bolding of titles

* import ExtraWorksheetHTML scss file

* Revert "fix: don't do a production build in dev mode"

This reverts commit b4aeeb5879d71a26172ddabbf7fee0f76a18d747.

epicfaace committed:
fix paths

epicfaace committed:
fix undefined element error (#1651)

epicfaace committed:
fix modal position and font size (#1653)

epicfaace committed:
remove margin for text edit cell, fixes #1609 (#1665)

yipenghe committed:
Fix several small issues (#1636)

* fix bundle permissions

* fix rerun

* newline for stdout & stderr in bundle detail

* disable r, u, t shortcuts when bundles are selected

* scope keyboard events for bulk bundle operations, rename status->state in bundle detail

Levilian committed:
cursor remains the same after refresh; signposting for refresh (#1629)

* cursor remains the same after refresh; signposting for refresh

epicfaace committed:
Control modal entirely with react props (#1671)

* control modal with react props, remove jquery, fixes #1670

* Update frontend/src/components/worksheets/Worksheet/Worksheet.js

Levilian committed:
markdown action tooltips & worksheet title untitled handling (#1689)

epicfaace committed:
Add travis checks for frontend style/formatting (#1673)

* format files using prettier

* add travis checks for frontend style/formatting

* Remove "cd frontend"

* add format to lint-staged

* fix format

* include proper command for lint-staged prettier

* git add

Levilian committed:
operations with shortcuts and their focus effects (#1649)

- smooth scrolling
- web terminal shortcuts link updated
- resolved focusIndex not updating properly issue
- shortcuts and their post-action focus effects implemented

epicfaace committed:
Remove text-align prop (#1650)

* fix text-align prop

* remove align attribute

yipenghe committed:
prettier format

yipenghe committed:
allow summary to have more space (#1691)

* allow summary to have more space, make width smaller for other columns

* remove unused styles

yipenghe committed:
Delete worksheet button (#1692)

* style delete worksheet, make it not too obvious

* prettier format

* add TODO

* change language a bit

yipenghe committed:
Update bundle detail during runs (#1675)

* update bundle detail when data changes

* persist selection when autorefresh

* set time out for fetching data every 3 second in bundle detail, remove checkboxes after command consistently

* add small border to Header

* setInterval, update fileBrowser data as well"

* indent

* prettier

* only keep fetching data for staged/starting/preparing/running bundles in bundle detail

epicfaace committed:
fix howto guides link (#1696)

yipenghe committed:
upload-cancel-upload works (#1720)

yipenghe committed:
redirect to home/dashboard when going to /worksheets link (#1719)

* redirect to home/dashboard when going to /worksheets link

* better if-else structure

yipenghe committed:
Use placeholder for empty worksheet titles (#1739)

* all value evaluated to false should use placehold

* Update frontend/src/components/EditableField.js

epicfaace committed:
fix footer height (#1740)

yipenghe committed:
remove context menu for bundle rows (#1705)

yipenghe committed:
Update shortcut help (#1718)

* keyboard shortcuts help update

yipenghe committed:
merge master to frontend

yipenghe committed:
merge master to frontend

yipenghe committed:
prettier formatting

epicfaace committed:
Fix multiple deletion and deletion focus issues, pressing down on empty worksheet (#1743)

* keep focus the same upon deletion

* wait for a single deletion to complete before a keyboard press can trigger another deletion

* fix bug with deleting last item

* fix: don't crash when pressing down on an empty worksheet

yipenghe committed:
a t/u/r

yipenghe committed:
Handle File cancel properly (#1747)

* upload-cancel

* remove stackoverflow link

* comment

epicfaace committed:
Show Loading sign when saving a worksheet item (#1693)

yipenghe committed:
Loading sign when refreshing large worksheets, keyboard help update, update sign for bundle commands (#1748)

Adds refresh indicator for large worksheets,
keyboard help update,
bulk operation execution indicator

epicfaace committed:
Change placeholder to Loading... (#1690)

* display title as Loading when info is not ready

yipenghe committed:
merge master into frontend

yipenghe committed:
use toast to notify executing command, change refresh color to blue (toast.info)

yipenghe committed:
add loading sign for executing command, toast sign for success

yipenghe committed:
update keyboard shortcut menu (#1755)

* update keyboard shortcut menu


* just use th style

yipenghe committed:
Add delete dialog when deleting markdown blocks; fix cycling table bug; shift+enter open bundle detail in new window (#1750)

* add deletion dialog

* warn about deletion, tips for multiple block deletion


* prevent cycling the first bundle table

* shift+enter open bundle detail website

* don't trigger delete when not focused'

* rename deleteItemCallback

teetone committed:
1753: fix dev environment. copy over table item logic from master (#1754)

* 1753: fix dev environment. copy over table item logic from master

* 1753: fixed styles with prettier

* 1753: run prettier in pre-commit
